lib/policies/userAgentRequestPolicyFactory.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+import { HttpPipelineLogLevel } from "../httpPipelineLogLevel";
+import { HttpRequest } from "../httpRequest";
+import { HttpResponse } from "../httpResponse";
+import { BaseRequestPolicy, RequestPolicy } from "../requestPolicy";
+import { RequestPolicyFactory } from "../requestPolicyFactory";
+import { RequestPolicyOptions } from "../requestPolicyOptions";
+
+/**
+ * Get a RequestPolicyFactory that creates UserAgentRequestPolicies.
+ * @param userAgent The userAgent string to apply to each outgoing request.
+ */
+export function userAgentRequestPolicyFactory(userAgent: string): RequestPolicyFactory {
+    return (nextPolicy: RequestPolicy, options: RequestPolicyOptions) => {
+        return new UserAgentRequestPolicy(userAgent, nextPolicy, options);
+    };
+}
+
+class UserAgentRequestPolicy extends BaseRequestPolicy {
+    constructor(private _userAgent: string, nextPolicy: RequestPolicy, options: RequestPolicyOptions) {
+        super(nextPolicy, options);
+    }
+
+    send(request: HttpRequest): Promise<HttpResponse> {
+        if (this.shouldLog(HttpPipelineLogLevel.INFO)) {
```

Comment:
what do we do in the browser where we can't set User-Agent? Probably not that important

Reply:
That's kind of outside of the scope of the request policy. That would be down in the HttpClient function where it would have to strip away the UserAgent header. What happens if you try to send a request with a UserAgent header in the browser? Does it throw an error, or does it just replace it for you?

Reply:
I think it ignores the value you specified and uses its own.

Reply:
Then I don't think we need to have extra logic for the browser then. If this essentially becomes a no-op, then I'm fine with leaving it.